internal/services/datadog/azurerm_datadog_monitor_sso_configuration.go

```diff
@@ -143,10 +164,13 @@ func resourceDatadogSingleSignOnConfigurationsRead(d *pluginsdk.ResourceData, me
 	if model := resp.Model; model != nil {
 		if props := model.Properties; props != nil {
 			// per the create func
-			singleSignOnEnabled := props.SingleSignOnState != nil && *props.SingleSignOnState == singlesignon.SingleSignOnStatesEnable
-			d.Set("single_sign_on_enabled", singleSignOnEnabled)
+			d.Set("single_sign_on", string(*props.SingleSignOnState))
```

Comment:
This could be nil and panic

```suggestion
			d.Set("single_sign_on", string(pointer.From(props.SingleSignOnState)))
```